pkg/p2p/libp2p/libp2p.go

```diff
@@ -55,19 +56,16 @@ type Service struct {
 }
 
 type Options struct {
-	PrivateKey  *ecdsa.PrivateKey
-	Overlay     swarm.Address
-	Addr        string
 	DisableWS   bool
 	DisableQUIC bool
-	NetworkID   uint64
 	Addressbook addressbook.Putter
 	Logger      logging.Logger
 	Tracer      *tracing.Tracer
 }
 
-func New(ctx context.Context, o Options) (*Service, error) {
-	host, port, err := net.SplitHostPort(o.Addr)
+func New(ctx context.Context, privateKey *ecdsa.PrivateKey, networkID uint64, overlay swarm.Address, addr string,
```

Comment:
Actually, private key is optional, it can be ephemeral in libp2p. I would put this back to Options, with the check if it is nil a few lines below.

Reply:
Hmm yes, my bad, I had trouble getting this key from libp2p in that case, and I forgot to tackle it later 🙈 nice catch!

Reply:
So,k I took another look at it. It seems that if the private key is not provided as one of the libp2p options, libp2p will use a default random identity then. This implementation seems to create a RSA key pair. What would you suggest in this case? I don't think those are interchangeable. Should we create a random private key in this case? Maybe I am missing something here, LTMWYT :)

Reply:
Yes, that is correct and I did not want to restrict this possibility, as libp2p is not required to have an identity option to work. Anyways, this is out of scope of this PR.

Reply:
So, you are OK with leaving it as required field for now?

Reply:
Well, no. I think that it should be in Options.

Reply:
Sorry, maybe I was not clear enough. What I am trying to say is, what key should I use for signing in this case, when private key is nil? If we don't specify the key to libp2p, it will create it's own random RSA key pair, which is of different type then what we are using in Signer, and might not be interchangeable.

Reply:
Ah ok, sorry, I confused swarm key with libp2p key 🙈 Should be fixed now!